pgx/src/lib.rs

```diff
@@ -284,29 +287,28 @@ macro_rules! pg_magic_func {
         #[allow(unused)]
         #[link_name = "Pg_magic_func"]
         pub extern "C" fn Pg_magic_func() -> &'static pgx::pg_sys::Pg_magic_struct {
+            use core::mem::size_of;
             use pgx;
-            use std::mem::size_of;
-            use std::os::raw::c_int;
 
             #[cfg(any(feature = "pg10", feature = "pg11", feature = "pg12"))]
             const MY_MAGIC: pgx::pg_sys::Pg_magic_struct = pgx::pg_sys::Pg_magic_struct {
-                len: size_of::<pgx::pg_sys::Pg_magic_struct>() as c_int,
-                version: pgx::pg_sys::PG_VERSION_NUM as c_int / 100,
-                funcmaxargs: pgx::pg_sys::FUNC_MAX_ARGS as c_int,
-                indexmaxkeys: pgx::pg_sys::INDEX_MAX_KEYS as c_int,
-                namedatalen: pgx::pg_sys::NAMEDATALEN as c_int,
-                float4byval: pgx::pg_sys::USE_FLOAT4_BYVAL as c_int,
-                float8byval: pgx::pg_sys::USE_FLOAT8_BYVAL as c_int,
+                len: size_of::<pgx::pg_sys::Pg_magic_struct>() as i32,
```

Comment:
This is mildly dangerous for portability... according to https://doc.rust-lang.org/std/os/raw/type.c_int.html, this is almost always an i32, but it's possible it is not... Ref: https://github.com/rust-lang/rfcs/issues/3012

Reply:
I *think* this will be fine as long as we don't end up with like `avr` or `msp430` users.